pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,438 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains utility functions for building boolean conditionals for noise models"""
+
+from functools import partial
+from inspect import isclass, signature
+
+import pennylane as qml
+from pennylane.boolean_fn import BooleanFn
+from pennylane.wires import Wires
+
+# pylint: disable = unnecessary-lambda, too-few-public-methods
+
+
+class WiresIn(BooleanFn):
+    """A ``Conditional`` for evaluating if a given wire exist in a specified set of wires
+
+    Args:
+        wires (Union[list[int, str], Wires]): sequence of wires for building the wire set.
+
+    .. seealso:: Users are advised to use :func:`~.wires_in` for a functional construction.
+    """
+
+    def __init__(self, wires):
+        self._cond = set(wires)
+        self.condition = self._cond
+        super().__init__(lambda x: _get_wires(x).issubset(self._cond), f"WiresIn({list(wires)})")
+
+
+class WiresEq(BooleanFn):
+    """A ``Conditional`` for evaluating if a given wire is equal to a specified set of wires
+
+    Args:
+        wires (Union[list[int, str], Wires]): sequence of wires for building the wire set.
+
+    .. seealso:: Users are advised to use :func:`~.wires_eq` for a functional construction.
+    """
+
+    def __init__(self, wires):
```

Comment:
I might be misremembering, but I thought I saw you show in the meeting the input argument of `WiresEq` being an operation?

Reply:
Oh it's in the `wires_in` and `wires_eq` function below :thinking: Why do we need operation input?

Reply:
This is just an additional support to make the constructor functions more flexible and easy to use for the users.  Especially in a scenario where they are trying to automate building noise models based on some set of operations they have in their pipeline, then they can still make these conditional without having to extract wires manually.